### ISSUE TITLE
Incompatibility with quarkus-hibernate-validator

### ISSUE THREAD

**Issue by rmanibus** [quarkiverse/quarkus-bucket4j#23]
project including quarkus-hibernate-validator and this plugin are failing with

```
2023-04-15 10:25:35,138 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:104)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:70)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        ... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 15 more
Caused by: java.lang.NullPointerException: Cannot invoke "jakarta.validation.Validator.forExecutables()" because the return value of "io.smallrye.config.validator.BeanValidationConfigValidator.getValidator()" is null
        at io.smallrye.config.validator.BeanValidationConfigValidator.validatePropertyValue(BeanValidationConfigValidator.java:203)
        at io.smallrye.config.validator.BeanValidationConfigValidator.validateProperty(BeanValidationConfigValidator.java:191)
        at io.smallrye.config.validator.BeanValidationConfigValidator.validateMappingInterface(BeanValidationConfigValidator.java:57)
        at io.smallrye.config.validator.BeanValidationConfigValidator.validateMapping(BeanValidationConfigValidator.java:39)
        at io.smallrye.config.ConfigMappings.getConfigMapping(ConfigMappings.java:124)
        at io.smallrye.config.SmallRyeConfig.getConfigMapping(SmallRyeConfig.java:411)
        at io.quarkus.deployment.steps.Bucket4jProcessor$gatherRateLimitCheck21582745.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.Bucket4jProcessor$gatherRateLimitCheck21582745.deploy(Unknown Source)
        ... 16 more

2023-04-15 10:25:35,138 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:104)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:70)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        ... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 15 more
Caused by: java.lang.NullPointerException: Cannot invoke "jakarta.validation.Validator.forExecutables()" because the return value of "io.smallrye.config.validator.BeanValidationConfigValidator.getValidator()" is null
        at io.smallrye.config.validator.BeanValidationConfigValidator.validatePropertyValue(BeanValidationConfigValidator.java:203)
        at io.smallrye.config.validator.BeanValidationConfigValidator.validateProperty(BeanValidationConfigValidator.java:191)
        at io.smallrye.config.validator.BeanValidationConfigValidator.validateMappingInterface(BeanValidationConfigValidator.java:57)
        at io.smallrye.config.validator.BeanValidationConfigValidator.validateMapping(BeanValidationConfigValidator.java:39)
        at io.smallrye.config.ConfigMappings.getConfigMapping(ConfigMappings.java:124)
        at io.smallrye.config.SmallRyeConfig.getConfigMapping(SmallRyeConfig.java:411)
        at io.quarkus.deployment.steps.Bucket4jProcessor$gatherRateLimitCheck21582745.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.Bucket4jProcessor$gatherRateLimitCheck21582745.deploy(Unknown Source)
        ... 16 more

2023-04-15 10:25:35,138 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:104)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:70)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        ... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 15 more
Caused by: java.lang.NullPointerException: Cannot invoke "jakarta.validation.Validator.forExecutables()" because the return value of "io.smallrye.config.validator.BeanValidationConfigValidator.getValidator()" is null
        at io.smallrye.config.validator.BeanValidationConfigValidator.validatePropertyValue(BeanValidationConfigValidator.java:203)
        at io.smallrye.config.validator.BeanValidationConfigValidator.validateProperty(BeanValidationConfigValidator.java:191)
        at io.smallrye.config.validator.BeanValidationConfigValidator.validateMappingInterface(BeanValidationConfigValidator.java:57)
        at io.smallrye.config.validator.BeanValidationConfigValidator.validateMapping(BeanValidationConfigValidator.java:39)
        at io.smallrye.config.ConfigMappings.getConfigMapping(ConfigMappings.java:124)
        at io.smallrye.config.SmallRyeConfig.getConfigMapping(SmallRyeConfig.java:411)
        at io.quarkus.deployment.steps.Bucket4jProcessor$gatherRateLimitCheck21582745.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.Bucket4jProcessor$gatherRateLimitCheck21582745.deploy(Unknown Source)
        ... 16 more
```

**Comment by rmanibus** [comment]
probably related to https://github.com/quarkusio/quarkus/issues/24540

**Comment by rmanibus** [comment]
@radcortez do you have an idea about this one ?

The configuration is working properly in the tests of this project, but whenever I try to add the plugin to another quarkus project, it is causing a NPE: 

`java.lang.NullPointerException: Cannot invoke "jakarta.validation.Validator.forExecutables()" `

Any reason what could cause this in the recorder ?

I also tried to replace the injection by 

```
ConfigProvider.getConfig().unwrap(SmallRyeConfig.class)
                .getConfigMapping(RateLimiterConfig.class)
```
with the same result

**Comment by radcortez** [comment]
Probably because the extension recorder methods are running before Bean Validation is set up. We changed this recently for Config to have its own validator and not depend on the main validator in https://github.com/quarkusio/quarkus/pull/32586.

Can you try with that PR and check if the problem persists?

**Comment by rmanibus** [comment]
@radcortez it is working with main, thank you !

**Comment by rmanibus** [comment]
resolved by quarkus 3.0.1.Final
